tests/testthat/test-scale-gradient.R

```diff
@@ -0,0 +1,18 @@
+context("scale_gradient")
+
+# Limits ------------------------------------------------------------------
+
+test_that("points outside the limits are plotted as NA", {
+  p <- ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width, color = Petal.Length) ) +
+          geom_point() +
+          scale_color_gradient2(low = muted("green"), high = muted("navy"),
+                                mid = "gray80",
+                                midpoint = 3, limits = c(2, 6),
+                                na.value = "orange")
+  p_data <- ggplot_build(p)$data[[1]]
```

Comment:
Can you please use `layer_data()` here

Reply:
And I think a made up data set would make the test simpler. Maybe something like this?

```R
library(ggplot2)
df <- data.frame(x = c(0, 1, 2))

p <- ggplot(df, aes(x, 1, fill = x)) +
  geom_col() + 
  scale_fill_gradient2(limits = c(-1, 1), na.value = "orange")

layer_data(p)$fill
```